wordle-app/pages/WordOfTheDayGame.vue

```diff
@@ -0,0 +1,290 @@
+            width="10"
+            :model-value="game.stats.averageGuessesPercent(game.maxAttempts)"
+            >
+            {{  game.stats.averageGuessesPercent(game.maxAttempts).toFixed(0) }}%
+            </v-progress-circular>
+            <br/>
+            <i class="text-caption">
+              Average Guesses
+            </i>
+          </v-col>
+        </v-row>
+        <v-btn variant="outlined" @click="game.startNewGame()">
+          <v-icon size="large" class="mr-2"> mdi-restart </v-icon> Restart Game
+        </v-btn>
+      </v-alert>
+      <v-card-title v-else>Word Of The Day Game</v-card-title>
+
+      <v-card-text class="d-flex  flex-row justify-end">
+        <v-chip color="secondary" class="mr-2" @click="showUserNameDialog=true">
+          <v-icon class="mr-2" > mdi-account </v-icon> {{ nameUserNameDialog }}
+        </v-chip>
+        <v-chip color="secondary" class="mr-2">
+          <v-icon class="mr-2"> mdi-timer </v-icon> {{ seconds }} s.
+        </v-chip>
+      </v-card-text>
+
+      <GameBoardGuess v-for="(guess, i) of game.guesses" :key="i" :guess="guess" />
+
+      <div class="my-10">
+        <Keyboard />
+      </div>
+
+      <v-row class="mb-5" justify="end">
+        <v-col  cols="12" sm="4">
+          <WordList v-if="game.gameState === GameState.Playing" v-model="showWordsList" />
+        </v-col>
+        <v-col  cols="12" sm="4">
+          <v-btn @click="game.submitGuess()" color="primary">
+            Guess!
+          </v-btn>
+        </v-col>
+        <v-col  cols="12" sm="4">
+          <v-btn color="primary" @click="router.push('/Leaderboard')">Leaderboard</v-btn>
+        </v-col>
+      </v-row>
+    </v-card>
+
+    <UserNameDialog @okay="nameDialogClosed()"  v-model:show="showUserNameDialog" v-model:userName="nameUserNameDialog" />
+    
+  </v-container>
+</template>
+
+<script setup lang="ts">
+import { Game, GameState } from "../scripts/game";
+import nuxtStorage from "nuxt-storage";
+import Axios from 'axios'
+import type { Player } from "../scripts/player";
+import UserNameDialog from "~/components/UserNameDialog.vue";
+
+const router = useRouter();
+const game = reactive(new Game());
+game.startNewGame();
+provide("GAME", game);
+
+// User name dialog
+const showUserNameDialog = ref(false);
+const nameUserNameDialog = ref("");
+
+function nameDialogClosed() {
+  showUserNameDialog.value = false;
+  startStopwatch();
+}
+
+var showWordsList = ref(false);
+
+onMounted(() => {
+
+  window.addEventListener("keyup", onKeyup);
+
+
+  // Assignment 3 Task 2
+  // Check if user name is stored in local storage
+  checkUserNameLocalStorage();
+  checkUserName();
+
+  if(!showUserNameDialog.value){
+    startStopwatch();
+  }
+
+});
+
+
+onUnmounted(() => {
+  window.removeEventListener("keyup", onKeyup);
+});
+
+async function getWordFromApi(): Promise<string> {
+  let wordUrl = "word/wordOfTheDay";
+  const response = await Axios.get(wordUrl);
+  return response.data;
+}
```

Comment:
```suggestion
async function getWordFromApi(): Promise<string> {
  let wordUrl = "word/wordOfTheDay";
  const response = await Axios.get(wordUrl);
  return response.data;
}
```
We moved this to being inside `game.ts` so you can remove it

Reply:
You are right.